frontend/packages/pipelines-plugin/src/utils/pipeline-filter-reducer.ts

```diff
@@ -14,9 +15,48 @@ export enum SucceedConditionReason {
   ConditionCheckFailed = 'ConditionCheckFailed',
 }
 
+export const pipelineRunStatus = (pipelineRun): runStatus => {
+  const conditions = _.get(pipelineRun, ['status', 'conditions'], []);
+  if (conditions.length === 0) return null;
+
+  const succeedCondition = conditions.find((c) => c.type === 'Succeeded');
+  if (!succeedCondition || !succeedCondition.status) {
+    return null;
+  }
+
+  const status =
+    succeedCondition.status === 'True'
+      ? runStatus.Succeeded
+      : succeedCondition.status === 'False'
+      ? runStatus.Failed
+      : runStatus.Running;
+
+  if (succeedCondition.reason && succeedCondition.reason !== status) {
+    switch (succeedCondition.reason) {
+      case SucceedConditionReason.PipelineRunCancelled:
+      case SucceedConditionReason.TaskRunCancelled:
+      case SucceedConditionReason.Cancelled:
+        return runStatus.Cancelled;
+      case SucceedConditionReason.PipelineRunStopping:
+      case SucceedConditionReason.TaskRunStopping:
+        return runStatus.Failed;
+      case SucceedConditionReason.CreateContainerConfigError:
+      case SucceedConditionReason.ExceededNodeResources:
+      case SucceedConditionReason.ExceededResourceQuota:
+      case SucceedConditionReason.PipelineRunPending:
+        return runStatus.Pending;
+      case SucceedConditionReason.ConditionCheckFailed:
+        return runStatus.Skipped;
+      default:
+        return status;
+    }
+  }
+  return status;
+};
+
 // Converts the PipelineRun (and TaskRun) condition status into a human readable string.
 // See also tkn cli implementation at https://github.com/tektoncd/cli/blob/release-v0.15.0/pkg/formatted/k8s.go#L54-L83
-export const pipelineRunStatus = (pipelineRun): string => {
+export const pipelineRunStatusTitle = (pipelineRun): string => {
```

Comment:
Now we have the condition check twice. If someone changes just one this could result in really strange results.

Can you call `pipelineRunStatus` here instead and then convert the `status: runStatus` (or even better `status: ComputedStatus`) to a string? That should be just a switch statement like this:

```tsx
export const pipelineRunStatusTitle = (pipelineRun): string => {
  const status = pipelineRunStatus(pipelineRun);
  if (!status) return '-';
  switch (status) {
    case runStatus.Cancelled:
      return i18next.t('pipelines-plugin~Cancelled');
    ...
    default:
      return status;
  }
}
```